### DISCUSSION TITLE
Feature request: System.Text.Json (alternative to Newtonsoft)

### DISCUSSION THREAD

**Discussion by Zettersten** [IEvangelist/azure-cosmos-dotnet-repository#134] [Ideas]
Not sure if lower dependencies can manage this request/feature but it’d be cool to see this AMAZING project support the newer System.Text.Json for deserialization/serialization and attributing (keys, partitions, etc).

perhaps even, a “bring your own serializer” solution. 

Either way - would be eager to hear your alls thoughts!

**Comment by IEvangelist** [comment]
Hi @Zettersten - this is something that we've thought about for sure. I believe that in the future, the official Cosmos DB .NET SDK will be porting to the new `System.Text.Json` serializer. I'm curious if they're going to expose some sort of "bring your own serializer" solution. Which I agree would be ideal. But we're a bit hands tied unfortunately. We cannot change it, unless they do as it's simply not expose to us.

**Comment by karpikpl** [comment]
isn't that possible already with https://github.com/Azure/azure-cosmos-dotnet-v3/commit/d08a9c305de8d37fc3b6a9b7ad1efeacd621662d ? and the ability to provide your own serializer?

**Reply by IEvangelist** [reply]
At the time of writing, it wasn't - but I believe you're correct @karpikpl

**Comment by boylec** [comment]
@IEvangelist I believe this has happened recently where LINQ honors STJ serializer (or something)

https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4332

I for one would be overjoyed if this package would embrace System.Text.Json by default
